src/System.Collections/ref/System.Collections.cs

```diff
@@ -311,26 +316,26 @@ public partial class List<T> : System.Collections.Generic.ICollection<T>, System
         object System.Collections.ICollection.SyncRoot { get { throw null; } }
         bool System.Collections.IList.IsFixedSize { get { throw null; } }
         bool System.Collections.IList.IsReadOnly { get { throw null; } }
-        object System.Collections.IList.this[int index] { get { throw null; } set { } }
+        object? System.Collections.IList.this[int index] { get { throw null; } set { } }
         public void Add(T item) { }
         public void AddRange(System.Collections.Generic.IEnumerable<T> collection) { }
         public System.Collections.ObjectModel.ReadOnlyCollection<T> AsReadOnly() { throw null; }
-        public int BinarySearch(int index, int count, T item, System.Collections.Generic.IComparer<T> comparer) { throw null; }
+        public int BinarySearch(int index, int count, T item, System.Collections.Generic.IComparer<T>? comparer) { throw null; }
         public int BinarySearch(T item) { throw null; }
-        public int BinarySearch(T item, System.Collections.Generic.IComparer<T> comparer) { throw null; }
+        public int BinarySearch(T item, System.Collections.Generic.IComparer<T>? comparer) { throw null; }
         public void Clear() { }
         public bool Contains(T item) { throw null; }
         public System.Collections.Generic.List<TOutput> ConvertAll<TOutput>(System.Converter<T, TOutput> converter) { throw null; }
         public void CopyTo(int index, T[] array, int arrayIndex, int count) { }
         public void CopyTo(T[] array) { }
         public void CopyTo(T[] array, int arrayIndex) { }
         public bool Exists(System.Predicate<T> match) { throw null; }
-        public T Find(System.Predicate<T> match) { throw null; }
+        public [System.Diagnostics.CodeAnalysis.MaybeNullAttribute]T Find(System.Predicate<T> match) { throw null; }
```

Comment:
This is wrong. This should've been a return attribute on the member. 

```cs
[return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
public T Find(System.Predicate<T> match) { throw null; }
```

I will debug GenAPI and fix it.

Reply:
Today it looks like this is happening here: https://github.com/dotnet/arcade/blob/b6107674719638ea0794f01bd32e3069de6388eb/src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs#L69

Reply:
Thanks for digging, @ericstj.

> Is the inline syntax invalid or just undesirable?
It seems like the syntax is invalid. It is failing to compile. It seems like a straight forward fix. I will fix it on my branch for teaching GenAPI understand nullability.

Reply:
Cool, looks like we already have an overload for WiteAttributes that accepts a string prefix, so that's the way to go. https://github.com/dotnet/arcade/blob/b6107674719638ea0794f01bd32e3069de6388eb/src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Attributes.cs#L106

Reply:
Yes, I've already fixed it locally and its working. Will push it to my branch.

Reply:
Fixed.